resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -86,6 +86,13 @@ private[spark] class Client(
     sparkConf.get(AM_CORES)
   }
 
+  private val driverResourceTypes: collection.immutable.Map[String, String] =
```

Comment:
why not just `Map[String, String]`?

Reply:
If I change it to Map, I have a compilation error because the call below of sparkConf.getAllWithPrefix(..).toMap returns an immutable Map.
Error:(91, 84) type mismatch;
 found   : scala.collection.immutable.Map[String,String]
 required: scala.collection.mutable.Map[String,String]

Reply:
Argh, this is another class that imports a different `Map` and overrides the predef one... not your fault, at some point I'll get bothered by this and clean it up.

Also this should be `amResources` since (i) it's not just for the driver and (ii) it's not a list of types.